benchexec/tools/deagle.py

```diff
@@ -21,13 +22,32 @@ def name(self):
         return "Deagle"
 
     def project_url(self):
-        return "https://github.com/Misasasa/Deagle"
+        return "https://github.com/thufv/Deagle"
+
+    def version_geq(self, version_str1, version_str2):
+        version1 = version_str1.split('.')
+        version2 = version_str2.split('.')
+        for i in range(len(version1)):
+            if int(version1[i]) > int(version2[i]):
+                return True
+            if int(version1[i]) < int(version2[i]):
+                return False
+        return True
 
     def version(self, executable):
         return self._version_from_tool(executable, arg="--version")
 
     def cmdline(self, executable, options, task, rlimits):
-        return [executable] + [task.property_file] + [task.single_input_file]
+        ver = self.version(executable)
```

Comment:
This means that your tool is started once for every run just in order to get the command-line. This is not even parallelized in BenchExec and could thus increase the time for starting your runs significantly. Thus it would be better to cache this.

There are many options for this, for example:
- Add `@functools.lru_cache` to the `version` method.
- Store the version (or the flag whether its old/new) in the `executable()` method and read it in `cmdline()`.